dvc/objects/checkout.py

```diff
@@ -43,88 +36,9 @@ def _remove(fs_path, fs, in_cache, force=False):
     fs.remove(fs_path)
 
 
-def _verify_link(cache, fs_path, link_type):
-    if link_type == "hardlink" and cache.fs.getsize(fs_path) == 0:
-        return
-
```

Comment:
Since we now verify things early with temp file that is not empty - this situation can no longer occur, so we are saving ourselves an API call here for every file.